File: samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/spots/conversationspot/CloseConversationHandler.java
package systems.zlink.samples.supportchat.server.support.spots.conversationspot;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.spots.ZLinkSpotActorRequestHandler;
import systems.zlink.samples.supportchat.server.support.actors.SupportUserActor;
import systems.zlink.samples.supportchat.shared.contracts.Messages;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public final class CloseConversationHandler
        implements ZLinkSpotActorRequestHandler<
                ConversationSpot,
                SupportUserActor,
                Messages.CloseConversationReq,
                Messages.CloseConversationRes> {
    @Override
    public CompletionStage<Messages.CloseConversationRes> handle(
            ConversationSpot spot,
            SupportUserActor actor,
            ZLinkMessageContext context,
            Messages.CloseConversationReq request) {
        return CompletableFuture.completedFuture(spot.close(actor, request));
    }
}

File: samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/infrastructure/zlink/spots/conversationspot/handlers/CloseConversationHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.handlers

import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.kotlin.ZLinkSuspendingSpotActorRequestHandler
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.actors.SupportUserActor
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.conversationspot.ConversationSpot
import systems.zlink.samples.kotlin.supportchat.shared.contracts.CloseConversationReq
import systems.zlink.samples.kotlin.supportchat.shared.contracts.CloseConversationRes

class CloseConversationHandler :
    ZLinkSuspendingSpotActorRequestHandler<
        ConversationSpot,
        SupportUserActor,
        CloseConversationReq,
        CloseConversationRes,
    > {
    override suspend fun handle(
        spot: ConversationSpot,
        actor: SupportUserActor,
        context: ZLinkMessageContext,
        request: CloseConversationReq,
    ): CloseConversationRes = spot.close(actor, request)
}
